near-lake-primitives/src/types/receipts.rs

```diff
@@ -0,0 +1,169 @@
+use crate::near_indexer_primitives::{
+    types::{AccountId, Balance, Gas},
+    views, CryptoHash, IndexerExecutionOutcomeWithReceipt,
+};
+use near_crypto::PublicKey;
+use near_primitives_core::serialize::{base64_format, dec_format};
+
+#[derive(Debug, Clone)]
+pub struct ExecutedReceipt {
+    pub receipt_id: CryptoHash,
+    pub receiver_id: AccountId,
+    pub predecessor_id: AccountId,
+    pub status: ExecutionStatus,
+    pub execution_outcome_id: CryptoHash,
+    pub logs: Vec<String>,
+}
+
+impl From<&IndexerExecutionOutcomeWithReceipt> for ExecutedReceipt {
+    fn from(outcome_with_receipt: &IndexerExecutionOutcomeWithReceipt) -> Self {
+        Self {
+            receipt_id: outcome_with_receipt.receipt.receipt_id,
+            receiver_id: outcome_with_receipt.receipt.receiver_id.clone(),
+            predecessor_id: outcome_with_receipt.receipt.predecessor_id.clone(),
+            execution_outcome_id: outcome_with_receipt.execution_outcome.id,
+            logs: outcome_with_receipt
+                .execution_outcome
+                .outcome
+                .logs
+                .iter()
+                .map(Clone::clone)
+                .collect(),
+            status: (&outcome_with_receipt.execution_outcome.outcome.status).into(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ExecutionStatus {
+    SuccessValue(Vec<u8>),
+    SuccessReceiptId(CryptoHash),
+    Failure(String),
+    Unknown,
```

Comment:
Can someone remind me in which case `ExecutionStatusView` might be `Unknown`, in which case?

cc @frol @telezhnaya

Reply:
It is internal nearcore status when the execution has not yet started. It is impossible to observe an execution outcome with such a status when it is included in a block. Consider removing it. Ideally, nearcore should better separate types instead of leaking impossible states to the outside world.

Reply:
It's almost how I thought it. I think I am going to make it `Pending` in the Lake's `ExecutionStatus` for `Receipt` for the included yet not executed ones, what do you think?

Reply:
From a user point of view, I would prefer to get a list of items that I can immediately deal with, and mixing included and executed receipts does not make it simpler as users will need to filter out not-yet-executed receipts. Thus, I am not supporting the idea of artificial execution status, especially given that execution status is a property of an ExecutionOutcome with logs and a result.

Reply:
@frol I mean I want to give users the same structure `Receipt` with `status = ExecutionStatus::Pending` and `logs: vec![]` when they request something like `block.receipts_included()`

Executed ones would be achievable via `block.receipts_executed()` I find it convenient from the developer's perspective. Most indexers work only with executed ones and don't care about the included unless they are executed. Makes sense to you?

Reply:
@khorolets trying to chime in here. It threw me off in the beginning about the differentiation between `block.receipts_included()` and `block.receipts_executed()`. 

I think `block.receipts_pending()` makes more sense to me than `block.receipts_included()`, if the only difference between `blocks.receipts_executed()` and `blocks.receipts_included()` is that receipts from `.receipts_included` have not executed yet and thus are in `pending` status.

Reply:
@roshaans Sorry it doesn't work. Receipts are included in the block only once (e.g. block 100) and they might be considered as pending but when you call this method in next block you won't get them there. Though they are still pending, but not included anymore and might be not executed yet.

Reply:
@roshaans I like the direction of your thinking re "pending".

@khorolets I want to propose considering `postponed_receipts`.